src/docfx/publish/PublishModelBuilder.cs

```diff
@@ -47,7 +50,8 @@ public void SetPublishItem(FilePath file, JObject? metadata, string? outputPath)
         public (PublishModel, Dictionary<FilePath, PublishItem>) Build()
         {
             var publishItems = new Dictionary<FilePath, PublishItem>();
-            foreach (var (url, sourcePath, monikers) in _publishUrlMapBuilder.GetPublishOutput())
+            foreach (var (url, sourcePath, monikers) in _publishUrlMap.GetAllFiles().Concat(_linkResolver.GetAdditionalResources())
```

Comment:
Consider computing the `url` and `monikers` in the loop body, also remove `PublishUrlMap.GetPublishOutput`?
```suggestion
            foreach (var sourcePath in _publishUrlMap.GetAllFiles().Concat(_linkResolver.GetAdditionalResources())
```